backend/views/retrieve_test_view.py

```diff
@@ -79,85 +101,167 @@ def retrieve_json_from_name_date(test_name, query_date_time, request_type):
         return return_dict
 
     if request_type == TEST_QUESTIONS:
-        # TODO jcherry write the logic to get question data
-        # After merging the API PRs
-        return_dict["questions"] = []
+        item_type_map = gen_item_map(query_date_time)
+        question_type_map = gen_question_map(query_date_time)
+        question_map = get_items(
+            item, item_type_map, question_type_map,
+            query_date_time, en_id, fr_id, QUESTION_CHILDREN_MAP)
+        return_map = gen_return_map(question_map)
+        return_dict["questions"] = return_map
         return return_dict
 
     # if it is not one of the above, then return nothing
     return {}
 
 
+def gen_return_map(question_map):
+    en_map, fr_map = test_map_to_language_map(question_map)
+    return {EN: en_map, FR: fr_map}
+
+
+def test_map_to_language_map(cur_map):
+    en_map = {}
+    fr_map = {}
+    # get and sort keys
+    keys = cur_map.keys()
+    keys = list(keys)
+    keys.sort()
+    # if the keys are just EN and FR, then return the values
+    if keys == [EN, FR]:
+        return cur_map[EN], cur_map[FR]
+    # otherwise parse through the keys
+    for key in keys:
+        child_type = cur_map[key][TYPE]
+        child_map = cur_map[key][MAP]
+        child_en, child_fr = test_map_to_language_map(child_map)
+        print(child_type)
+        print(child_en)
+        print(child_fr)
+        # add the key as id if it these are dicts
+        if isinstance(child_en, dict):
+            child_en["id"] = key
+        if isinstance(child_fr, dict):
+            child_fr["id"] = key
+        en_map = add_to_map(child_type, child_en, en_map)
+        fr_map = add_to_map(child_type, child_fr, fr_map)
+    return en_map, fr_map
+
+
+def add_to_map(child_type, child_language_map, language_map):
+    if child_type in SINGLE_RETURN:
+        language_map[child_type] = child_language_map
+    elif child_type in language_map.keys():
+        language_map[child_type].append(child_language_map)
+    else:
+        language_map[child_type] = [child_language_map]
+    return language_map
+
+
+def get_items(parent_item, item_type_map, question_type_map, query_date_time,
+              en_id, fr_id, children_map):
+    return_map = {}
+    # get the parent id, get the type to determine how to handle it
+    parent_id, parent_type = get_item_type(
+        parent_item, item_type_map, question_type_map, query_date_time)
+    try:
+        children_types = children_map[parent_type]
+    except KeyError:
+        children_types = []
+    children_items = get_items_by_parent_id(parent_id, query_date_time)
+    for child in children_items:
+        _, child_type = get_item_type(

```

Comment:
Do you need the "_," before the _child_type_?

Reply:
there's 2 returns from the function; in this instance I only need the second, so I'm ignoring the first